test/e2e/retry_test.go

```diff
@@ -0,0 +1,53 @@
+//go:build executor
+// +build executor
+
+package e2e
+
+import (
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/suite"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+
+	wfv1 "github.com/argoproj/argo-workflows/v3/pkg/apis/workflow/v1alpha1"
+	"github.com/argoproj/argo-workflows/v3/test/e2e/fixtures"
+)
+
+type RetryTestSuite struct {
+	fixtures.E2ESuite
+}
+
+func (s *RetryTestSuite) TestRetryLimit() {
+	s.Given().
+		Workflow(`
+metadata:
+  generateName: test-backoff-
+spec:
+  entrypoint: main
+  templates:
+    - name: main
```

Comment:
How does this test work?

Reply:
the workflow fails with an exit code 1 and then initiates a retry, since the `limit` in the `retryStrategy` is set to 0, the workflow fails instantly and logs `No more retries left` as message